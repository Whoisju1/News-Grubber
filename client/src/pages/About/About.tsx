import React from 'react'
import styled, { css } from 'styled-components';

const StyledAbout = styled.article`
  grid-column: 1/-1;
  grid-row: 1/-1;
  display: grid;
  grid-template-columns: calc(100vw - 96%) 1fr calc(100vw - 96%);
  word-wrap: break-word;
  &::selection {
    background-color: var(--primary-color);
  }
  &::-moz-selection {
    background-color: var(--primary-color);
  }
  a {
    &,
    &:link,
    &:active,
    &:visited {
      color: var(--primary-color);
    }
  }
  h1 {
    font-weight: 600;
    font-size: 4rem;
    margin: 0 0 1rem 0;
    color: var(--primary-color);
  }

  h2 {
    font-weight: 600;
    font-size: 3rem;
    margin: 2rem 0 1rem 0;
    color: var(--primary-color);
  }

  p {
    text-align: justify;
  }

  section {
    grid-column: 2/3;
      p.bio {
        line-height: 2rem;
        &::after {
          content: "";
          clear: both;
          display: table;
        }
        img {
          max-height: 100%;
          height: auto;
          width: 15rem;
          object-fit: contain;
          filter: grayscale(80%);
          float: left;
          shape-outside: circle(50%);
          margin-right: 3rem;
          clip-path: circle(50%);
        }
      }
      &.site {
        ul {
          display: grid;
          grid-row-gap: .5rem;
        }
      }
    }
  section {
    margin: 3rem 0;
  }
`;

const About = () => {
  return (
    <StyledAbout>
      <section className="site">
        <h1>About NewsGrubber</h1>
        <p>
          NewsGrubber is a noncommercial site built only for experimentation
          and to put into practice some of what I've learned.
          It it scrapes the homepage of <a href="www.cnet.com" target="_blank">CNET.com</a> capturing the titles, subtitles, url,
          picture and other information of some of the displayed articles.
        </p>
        <h2>Tech Used to Build NewsGrubber</h2>
        <ul>
          <li>React and Typescript</li>
          <li>Node and Express</li>
          <li>MongoDB</li>
          <li>Docker</li>
          <li>Jest</li>
          <li>JWT for authentication</li>
          <li>Styled-Components (Styling)</li>
          <li>Git (Version Control)</li>
          <li>NPM (Package Management)</li>
          <li></li>
        </ul>
      </section>
      <section className="dev">
        <h1>About Developer</h1>
        <p className="bio">
          <img src={`${process.env.PUBLIC_URL}/images/dev_photo.jpg`} alt="developer"/>
          Hi, my name's Juan John-Charles. I'm a full stack web developer who love what he does.
          I'm originally from the country of Dominica, not Dominican Republic, just
          Dominica. It's a very tiny Island in the Caribbean, known as the "Nature Isle' of the Caribbean".
          It's a very beautiful place. I moved to the US and
          and fell in love with programming. I took my baby steps in programming at the Rutgers University
          Coding Bootcamp, where I learned the basics of web development but has since built upon the foundation that was laid, through taking
          courses, lots of personal practice, and professional experience. I make it a goal to grow, learn something new everyday,
          deepening my knowledge and sharpening my skills.
        </p>
      </section>
    </StyledAbout>
  );
}

export default About;
