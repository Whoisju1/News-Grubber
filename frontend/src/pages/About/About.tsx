import React from 'react'
import styled from 'styled-components';

const Section = styled.section`
  /* ... */
`;

function About() {
  return (
    <Section>
      About
    </Section>
  );
}

export default About;
