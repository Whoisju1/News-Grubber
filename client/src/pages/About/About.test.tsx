import React from 'react';
import About from './About';
import { shallow } from 'enzyme';

describe('About', () => {
  const setup = () => shallow(<About />);
  it('should match snapshot', () => {
    const wrapper = setup();
    expect(wrapper.html()).toMatchSnapshot();
  });
})
