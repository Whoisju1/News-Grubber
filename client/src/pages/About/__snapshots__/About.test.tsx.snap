// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About should match snapshot 1`] = `"<article class=\\"sc-bdVaJa fRabYq\\"><section class=\\"site\\"><h1>About NewsGrubber</h1><p>NewsGrubber is a noncommercial site built only for experimentation and to put into practice some of what I&#x27;ve learned. It it scrapes the homepage of <a href=\\"www.cnet.com\\" target=\\"_blank\\">CNET.com</a> capturing the titles, subtitles, url, picture and other information of some of the displayed articles.</p><h2>Tech Used to Build NewsGrubber</h2><ul><li>React and Typescript (Frontend)</li><li>Node and Express (Backend)</li><li>MongoDB (Database)</li><li>Docker</li><li>AWS Elastic Beanstalk (Infrastructure)</li><li>Travis (CI/CD)</li><li>Jest (Unit Tests)</li><li>Json Web Tokens(Authentication/Authorization)</li><li>Styled-Components (CSS Styling)</li><li>Git &amp; Github (Version Control)</li><li>NPM (Package Management)</li></ul></section><section class=\\"dev\\"><h1>About Developer</h1><p class=\\"bio\\"><img src=\\"/images/dev_photo.jpg\\" alt=\\"developer\\"/>Hi, my name&#x27;s Juan John-Charles. I&#x27;m a full stack web developer who love what he does. I&#x27;m originally from the country of Dominica, not Dominican Republic, just Dominica. It&#x27;s a very tiny Island in the Caribbean, known as the &quot;Nature Isle&#x27; of the Caribbean&quot;. It&#x27;s a very beautiful place. I moved to the US and and fell in love with programming. I took my baby steps in programming at the Rutgers University Coding Bootcamp, where I learned the basics of web development but has since built upon the foundation that was laid, through taking courses, lots of personal practice, and professional experience. I make it a goal to grow, learn something new everyday, deepening my knowledge and sharpening my skills.</p></section></article>"`;
